### PR TITLE
Switch to fs2-data-xml

### PR DESCRIPTION
**Pull request by rossabaker**
Replaces the SAX parser with fs2-data-xml.  This is a continuation of https://github.com/http4s/http4s-scala-xml/pull/25.

### CODE REVIEW

File: build.sbt [hunk 1]
```diff
@@ -1,21 +1,22 @@
...
-ThisBuild / crossScalaVersions := Seq("2.12.15", Scala213, "3.1.2")
+ThisBuild / crossScalaVersions := Seq("2.12.16", Scala213, "3.1.2")
 ThisBuild / scalaVersion := Scala213
 
-lazy val root = project.in(file(".")).aggregate(scalaXml).enablePlugins(NoPublishPlugin)
+lazy val root = tlCrossRootProject.aggregate(scalaXml)
 
 val http4sVersion = "0.23.12"
 val scalaXmlVersion = "2.1.0"
+val fs2DataVersion = "1.4.0"
```
Comment (ybasket) [inline build.sbt:14]:
```suggestion
val fs2DataVersion = "1.5.0"
```

That should the charset-related exception the tests caught before.

**Comment by armanbilge** [comment]
@ybasket wdyt about just making this branch into the new `main` and resetting the versioning to start from 0.1 or whatever?

**Comment by ybasket** [comment]
> @ybasket wdyt about just making this branch into the new `main` and resetting the versioning to start from 0.1 or whatever?

@armanbilge sounds good to me.

File: build.sbt [hunk 2]
```diff
@@ -1,21 +1,28 @@
 ThisBuild / tlBaseVersion := "0.23"
```
Comment (armanbilge) [inline build.sbt:1]:
```suggestion
ThisBuild / tlBaseVersion := "0.0"
```

**Comment by armanbilge** [comment]
Hum, seems I don't have the capability to change the default branch. @rossabaker? 😅

**Comment by ybasket** [comment]
Merging this now to not block further development, we can still swap branches later on.

**Comment by rossabaker** [comment]
Sorry, I'm confused by the branch switch request.  What needs to go where?

**Comment by armanbilge** [comment]
@rossabaker we want to make `main` the default branch, instead of `series/0.23`. Indeed, we want to reset the versioning and start from 0.1.

**Comment by rossabaker** [comment]
Branch is switched.

**Pull request merged**